### ISSUE TITLE
Unable to resolve peer dependency with latest version of React (react@17.0.1)

### ISSUE THREAD

**Issue by loewj** [JedWatson/react-select#4371]
### Description
Using the latest version of React (17.0.1 via Create React App), the dependency tree cannot be resolved. 

### Steps to Reproduce
The issue can be reproduced by simply making a new CRA project and installing the package:

`npx create-react-app my-app`
`cd my-app`
`npm i react-select`

When installing the react-select, you will receive the following error:

```zsh
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: my-app@0.1.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"^17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.8.0" from react-select@3.1.1
npm ERR! node_modules/react-select
npm ERR!   react-select@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/joshualoew/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/joshualoew/.npm/_logs/2021-01-11T19_49_59_206Z-debug.log
```

Here is what the package.json file looks like for a clean project:

```json
{
  "name": "my-app",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.11.8",
    "@testing-library/react": "^11.2.3",
    "@testing-library/user-event": "^12.6.0",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "react-scripts": "4.0.1",
    "web-vitals": "^0.2.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  }
}
```

Thank you, and please let me know if I can provide any more information!

**Comment by mikepage** [comment]
Use yarn, this works 👍

**Comment by Methuselah96** [comment]
This should be resolved as of [`react-select@3.2.0`](https://github.com/JedWatson/react-select/releases/tag/react-select%403.2.0).

**Comment by loewj** [comment]
Thank you, @Methuselah96!
